src/dbSta/src/dbNetwork.cc

```diff
@@ -3035,14 +3037,44 @@ void dbNetwork::hierarchicalConnect(dbITerm* source_pin,
   // in hierarchy, which is ok, and the source/dest modnet will be null
   dbModNet* source_db_mod_net = source_pin->getModNet();
   dbModNet* dest_db_mod_net = dest_pin->getModNet();
-  // case 1: source/dest in same module
```

Comment:
Its a bit odd to remove the case 1 comment while keeping the case 2 comment.

Reply:
Agreed. To help that section of the code has been re-commented. Please note a comment about why it is possible that a dest_pin might be apparently connected to another flat net, other than the possibly new source flat net, has been added (which will hopefully explain your prior observation).